pkg/util/push/otel.go

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package push
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	kitlog "github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/tenant"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/weaveworks/common/middleware"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/util/log"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+const (
+	pbContentType   = "application/x-protobuf"
+	jsonContentType = "application/json"
+
+	otelParseError = "otlp_parse_error"
+	maxErrMsgLen   = 1024
+
+	messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"
```

Comment:
would be nice if the error message shows which of the two numbers is the message size and which is the max.
I know that it could be guessed based on which is larger, but I'd prefer to not guess.

Reply:
```
❯ ag "received message larger than max"
vendor/google.golang.org/grpc/rpc_util.go
570:		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max length allowed on current machine (%d vs. %d)", length, maxInt)
573:		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max (%d vs. %d)", length, maxReceiveMessageSize)

pkg/util/push/otel.go
34:	messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"

pkg/util/http.go
29:const messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"
```

I wonder if there is a reason why we follow what `grpc` is printing. Do we do an error comparision somewhere?

And if I change this, should I change the existing error message too in `pkg/util/http.go`?

Reply:
I've made this change right now, but will revert it if you think otherwise.